### ISSUE TITLE
Slow response and out of heap space

### ISSUE THREAD

**Issue by scott-ywm** [freeplane/freeplane#1146]
**Describe the bug**
Freeplane responds very slowly after it's upgraded to 1.11.2. For example, it would take several minutes even if you resize the app windows and move the window to an extended screen. Below please check the error log for this operation.

**To Reproduce**

**Expected behavior**

**Mind map files**
If possible, upload a zipped mind map file where the problem occurs.

**Screenshots**

**Desktop (please complete the following information):**
 - Java Version

**Additional context**
STDERR: Exception in thread "Image Fetcher 0"
STDERR: java.lang.OutOfMemoryError: Java heap space
STDERR: Exception in thread "Image Fetcher 0"
STDERR: java.lang.OutOfMemoryError: Java heap space
STDERR: Exception in thread "AWT-EventQueue-0"
STDERR: java.lang.OutOfMemoryError: Java heap space
STDERR: Exception in thread "AWT-EventQueue-0"
STDERR: java.lang.OutOfMemoryError: Java heap space
STDERR: Exception in thread "Image Fetcher 0"
STDERR: java.lang.OutOfMemoryError: Java heap space
STDERR: Exception in thread "Image Fetcher 0"
STDERR: java.lang.OutOfMemoryError: Java heap space
STDOUT: May 15, 2023 4:20:45 PM org.freeplane.core.util.LogUtils info.

**Comment by dpolivaev** [comment]
The maps available to me work fine. Therefore I need an example of the map where you experience the problem in order to investigate it. You can use an addon mentioned in https://github.com/freeplane/freeplane/discussions/710 and the older Freeplane version if you want to remove any content. However, if the problem is related to the images, they should be attached.

**Comment by scott-ywm** [comment]
Thanks for your prompt response!
I installed the add-on and tried to obfustrate the map however it failed. Enclosed please check the screen shot.

Best regards,

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Comment by dpolivaev** [comment]
No screen shot found.

**Comment by scott-ywm** [comment]
Attached, it’s just said error executing the script: Java heap space.
Although the script failed there’s a duplicated map which is enclosed too.

Best,

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Comment by scott-ywm** [comment]
You're right. It seems something is wrong with my map. I tried to move each branch to other files and re-assembled them in a new map. Then the problem was solved however I lost the cross-branch link

**Comment by dpolivaev** [comment]
Actually, if there are any conditions where Freeplane behaves as you described, it is a bug that can be reproduced under such conditions. And this bug is likely to affect other people and other maps too. If you could send me the original map privately I would appreciate it and fix the issue. You can write me personally to 

dimitry (at) freeplane (dot) org

**Comment by scott-ywm** [comment]
OK

**Comment by dpolivaev** [comment]
Thanks, I got your mind map. It looks like you do not use formulas and scripts, and you have many images. Unfortunately, the images were not attached to the email. Without the images, I could not see any high memory consumption or delays. So. if you could mail me the images and your auto.properties configuration file I could continue my analysis.

-----------------

If Freeplane runs out of memory and becomes slow, the issues actually might be related. So I want to describe which checks you can do locally.

You can check if Freeplane option Preferences…->Environment->Cache->for images selects "use disk", it is the recommended option.

You can increase the maximum memory available to Freeplane from the default 512m to 1g or even higher depending on how much RAM you have available. Follow the instructions available at https://docs.freeplane.org/attic/old-mediawiki-content/Troubleshooting.html#freeplane-throws-out-of-memory-exception.

Please let me know if any of these recommendations help in your case.

**Comment by dpolivaev** [comment]
You can also try to either copy your mind map file to another directory and open it there so that the images can not be found, or check your image file sizes. If the files are too big it can cause the problems you described.

**Comment by scott-ywm** [comment]
Thanks! I checked the option and it's the Cache "use disk". And I moved the map to another directory and it works, with no error during resizing. Most likely the problem is the number of images, although the image size is not big, below 3M each. 
However it's interesting as I said before, I moved all the first level branches and then reassembled them in a new map, it works fine as well with all the images as before, I only lost the connections between branches.
